### ISSUE TITLE
Lightgbm cv feature importance python 

### ISSUE THREAD

**Issue by MotoRZR** [microsoft/LightGBM#1445]
It would be useful for the cv function to retain mean/std of feature importances across the cv folds.  I don’t believe this is currently implemented.

**Comment by guolinke** [comment]
you can do this by yourself simply, just get Booster from cv folds, and then calculate mean/std.

**Comment by MotoRZR** [comment]
Hi, thanks for the quick response.  The LightGBM native CV function I believe just returns a dictionary of results rather than the individual booster parameters.  Or am I missing something.

**Comment by guolinke** [comment]
@StrikerRUS can we return the `cvfolds` in lgb.cv as well ?

**Comment by StrikerRUS** [comment]
@guolinke Did you mean `return dict(results), cvfolds` tuple instead of
https://github.com/Microsoft/LightGBM/blob/3f401477872813a81ca98e90b9b42d085b1013d2/python-package/lightgbm/engine.py#L469

It'll be breaking changes. Maybe push `cvfolds` into the dict to save the current method signature?

**Comment by aerdem4** [comment]
@guolinke @StrikerRUS It would be nice to have an "ensemble of models" object. The benefits of having such a class is to return importances and predictions as an average. Currently, especially on Kaggle, people always do the cross-validation and ensembling at the same time. cv function can either return all generated models or out of sample predictions and average ensembled predictons for the test set if given. If you think that this feature is nice to have, I would like to help.

**Comment by JoshuaC3** [comment]
@StrikerRUS could we not do something like,

```
if return_feature_importances:
    return dict(results), dict(agg_feaure_importances)
else:
    return dict(results)
```

with default as `None` or `False`. Default behaviour then doesn't break.

**Comment by StrikerRUS** [comment]
@JoshuaC3 I suppose it's possible. Would you mind to create a PR?

**Comment by StrikerRUS** [comment]
Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.

**Comment by StrikerRUS** [comment]
Implemented in #3204.
Now it is possible to get feature importances from `CVBooster`.
